chan/eb_chan.h

```diff
@@ -24,6 +24,14 @@
 
 #include <stddef.h>
 #include <stdbool.h>
+
+#if defined(__MINGW32__) || defined(__CYGWIN__)
```

Comment:
```suggestion
#if defined(__MINGW32__) || defined(__CYGWIN__)
```
Judge WIN32 and POSIX?
```
#ifdef _WIN32
#if !defined(_POSIX_C_SOURCE) \
&& !defined(_POSIX1_SOURCE) \
&& !defined(_POSIX_SOURCE)
```